app/cmd/docker_run.go

```diff
@@ -0,0 +1,191 @@
+package cmd
+
+import (
+	"archive/tar"
+	"bytes"
+	"context"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strconv"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/container"
+	"github.com/docker/docker/client"
+	"github.com/docker/go-connections/nat"
+	"github.com/jenkins-zh/jenkins-cli/app/i18n"
+	"github.com/spf13/cobra"
+)
+
+func init() {
+	rootCmd.AddCommand(dockerRunCmd)
+	dockerRunCmd.Flags().StringVarP(&dockerRunOptions.ImageName, "image-name", "", "test",
+		i18n.T("name of the image which contains upgraded jenkins and plugins"))
+	dockerRunCmd.Flags().StringVarP(&dockerRunOptions.IP, "docker-ip", "", "127.0.0.1",
+		i18n.T("the ip address of the computer you want to use"))
+	dockerRunCmd.Flags().StringVarP(&dockerRunOptions.Tag, "tag", "", "latest",
+		i18n.T("the tag of the images"))
+	dockerRunCmd.Flags().IntVarP(&dockerRunOptions.DockerPort, "docker-port", "", 2375,
+		i18n.T("the port to connect to docker"))
+	dockerRunCmd.Flags().StringVarP(&dockerRunOptions.DockerfilePath, "dockerfile-path", "", "/tmp/Dockerfile",
+		i18n.T("where you want the dockerfile to be placed"))
+	dockerRunCmd.Flags().IntVarP(&dockerRunOptions.JenkinsPort, "jenkins-port", "", 8081,
+		i18n.T("The port you want to used to connect to jenkins"))
+	dockerRunCmd.Flags().StringVarP(&dockerRunOptions.WarPath, "war-path", "", "",
+		i18n.T("where your war is placed"))
+}
+
+//DockerRunOptions contains some of the options used to create a docker image and run a container
+type DockerRunOptions struct {
+	ImageName      string
+	Tag            string
+	IP             string
+	DockerPort     int
+	DockerfilePath string
+	JenkinsPort    int
+	WarPath        string
+}
+
+var dockerRunOptions DockerRunOptions
+
+var dockerRunCmd = &cobra.Command{
+	Use:   "docker run",
+	Short: i18n.T("Create a image and start a container in docker where all upgraded plugins and jenkins run in order to test their eligibility"),
+	Long: i18n.T("This command is used to create a image and start a container, where all upgraded plugins and jenkins run in. This command relies on the war file provided by Custom War Packager.\n" +
+		"The war can be made by calling 'jcli cwp --config-path <yamlfilename>' and the yaml file will be created using the command 'jcli create yaml'."),
+	PreRunE: dockerRunOptions.createDockerfile,
+	RunE:    dockerRunOptions.createImageAndRunContainer,
+	Example: `jcli docker run --docker-ip <ip> --docker-port <port> --war-path <pathToWar>`,
+}
+
+//GetDockerIPAndPort returns a string contains IP and port of a local or remote host
+func (o *DockerRunOptions) GetDockerIPAndPort() string {
+	ip := o.IP
+	port := o.DockerPort
+	return fmt.Sprintf("tcp://%s:%d", ip, port)
+}
```

Comment:
I saw that this function is useless, make sure to keep this function?

Reply:
All right. I've deleted this function and changed the .sh script to start jenkins without setupWizard.